builtin/credential/cert/path_login.go

```diff
@@ -155,7 +155,7 @@ func validateConnState(roots *x509.CertPool, cs *tls.ConnectionState) ([][]*x509
 		KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
 	}
 	certs := cs.PeerCertificates
-	if len(certs) == 0 {
+	if certs == nil || len(certs) == 0 {
```

Comment:
I looked through the history of the file and looked at line 158 in each revision. Based on that, this is almost certainly already fixed in cce15445c979cc74f26007dddcd4a0fbfe42ed0b, so feel free to close it.